### PR TITLE
Added buttons in fault display to open relevant fault panels

### PR DESCRIPTION
**Pull request by Derikka**

### CODE REVIEW

File: cavityFaultDisplay.py [hunk 1]
```diff
@@ -74,6 +83,24 @@ def __init__(self, parent=None, args=None, macros=None):
             horizontalLayout.addWidget(shortDescriptionLabel)
             horizontalLayout.addWidget(statusLabel)
 
+            if fault.button_level == "GUI":
+                button = PyDMRelatedDisplayButton(filename=fault.button_command)
+
+            elif fault.button_level == "EDM":
+                button = PyDMEDMDisplayButton()
+                button.filenames = [fault.button_command]
+                button.macros = fault.macros + ',' + fault.button_macro
```
Comment (Derikka) [inline cavityFaultDisplay.py]: This now successfully opens EDM panels to their specific tab. For panels that don't have a fault.button_macro (aka a column in the "Button Macros" cvs spreadsheet), will this cause any problems? With the way I implemented this, there will always be a ',' added onto button.macros. It doesn't look appear to have broken anything, but I just wanted to point it out

File: BackEnd/codeSnippet.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+
+example_pvPrefix = "ACCL:L2B:0920:"
+namePV = PV(example_pvPrefix + DESCRIPTION_SUFFIX)
+
+# Arbitrarily picking index 0 which should be "Parked"
+print(faults[0])
+namePV.put(faults[0])
+print(namePV.get())
```
Comment (lisazacarias) [inline BackEnd/codeSnippet.py]: Is this a test file? We can probably actually make this a unit test! I can show you how!

File: cavityFaultDisplay.py (hunk 1 above)
Comment (lisazacarias) [reply cavityFaultDisplay.py]: Ok good to know! We can keep an eye on it! Alternatively, you can do 
```suggestion
                button.macros = fault.macros + (',' + fault.button_macro if fault.button_macro else "")
```

File: cavityFaultDisplay.py [hunk 3]
```diff
@@ -74,6 +83,24 @@ def __init__(self, parent=None, args=None, macros=None):
...
+
+            elif fault.button_level == "PyDM":
+                button = PyDMRelatedDisplayButton()
+                button.filenames = [fault.button_command]
+
+            else:
```
Comment (lisazacarias) [inline cavityFaultDisplay.py:91]: It looks like GUI, PyDM, and other are all the same, so I'd probably just do if EDM, else

File: BackEnd/codeSnippet.py (hunk 2 above)
Comment (Derikka) [reply BackEnd/codeSnippet.py]: Hmm, this is something leftover from Gabby's code. I didn't realize it was even here!

Comment (lisazacarias) [reply BackEnd/codeSnippet.py]: Oh then we can probably delete it!

Comment (Derikka) [reply BackEnd/codeSnippet.py]: Deleted!

File: cavityFaultDisplay.py (hunk 1 above)
Comment (Derikka) [reply cavityFaultDisplay.py]: Ooh, thank you!!! I was hoping you would have a clean coding fix and this is perfect, thank you:D

File: cavityFaultDisplay.py (hunk 3 above)
Comment (Derikka) [reply cavityFaultDisplay.py:91]: Okay, updated!

File: cavityFaultDisplay.py (hunk 1 above)
Comment (Derikka) [reply cavityFaultDisplay.py]: Tested it on an OPI and it works:D

**Review by lisazacarias — APPROVED** [review]

**Pull request merged**